From c6f47cc7b9245add7db1d841e49c6bf0bb917a6c Mon Sep 17 00:00:00 2001
From: Zumeng Chen <zumeng.chen@windriver.com>
Date: Thu, 20 Apr 2017 14:28:23 +0800
Subject: [PATCH 345/347] clocksource: keystone: fix compile warnings

Set the right function since these many warnings really needs return
something big.

drivers/clocksource/timer-keystone.c: In function 'keystone_timer_init':
drivers/clocksource/timer-keystone.c:161:10: warning: 'return' with a value, in function returning void
   return -EINVAL;
          ^
drivers/clocksource/timer-keystone.c:150:13: note: declared here
 static void keystone_timer_init(struct device_node *np)
             ^~~~~~~~~~~~~~~~~~~
drivers/clocksource/timer-keystone.c:167:10: warning: 'return' with a value, in function returning void
   return -ENXIO;
          ^
drivers/clocksource/timer-keystone.c:150:13: note: declared here
 static void keystone_timer_init(struct device_node *np)
             ^~~~~~~~~~~~~~~~~~~
drivers/clocksource/timer-keystone.c:174:10: warning: 'return' with a value, in function returning void
   return PTR_ERR(clk);
          ^~~~~~~~~~~~
drivers/clocksource/timer-keystone.c:150:13: note: declared here
 static void keystone_timer_init(struct device_node *np)
             ^~~~~~~~~~~~~~~~~~~
drivers/clocksource/timer-keystone.c:227:9: warning: 'return' with a value, in function returning void
  return 0;
         ^
drivers/clocksource/timer-keystone.c:150:13: note: declared here
 static void keystone_timer_init(struct device_node *np)
             ^~~~~~~~~~~~~~~~~~~
drivers/clocksource/timer-keystone.c:231:9: warning: 'return' with a value, in function returning void
  return error;
         ^~~~~
drivers/clocksource/timer-keystone.c:150:13: note: declared here
 static void keystone_timer_init(struct device_node *np)
             ^~~~~~~~~~~~~~~~~~~
In file included from ./include/linux/clocksource.h:18:0,
                 from ./include/linux/clockchips.h:13,
                 from drivers/clocksource/timer-keystone.c:15:
drivers/clocksource/timer-keystone.c: At top level:
./include/linux/of.h:1004:20: warning: comparison of distinct pointer types lacks a cast
        .data = (fn == (fn_type)NULL) ? fn : fn  }
                    ^
./include/linux/of.h:1020:3: note: in expansion of macro '_OF_DECLARE'
   _OF_DECLARE(table, name, compat, fn, of_init_fn_1_ret)
   ^~~~~~~~~~~
./include/linux/clocksource.h:247:2: note: in expansion of macro 'OF_DECLARE_1_RET'
  OF_DECLARE_1_RET(clksrc, name, compat, fn)
  ^~~~~~~~~~~~~~~~
drivers/clocksource/timer-keystone.c:234:1: note: in expansion of macro 'CLOCKSOURCE_OF_DECLARE'
 CLOCKSOURCE_OF_DECLARE(keystone_timer, "ti,keystone-timer",
 ^~~~~~~~~~~~~~~~~~~~~~
  LD      drivers/cpuidle/governors/built-in.o
drivers/clocksource/timer-keystone.c: In function 'keystone_timer_init':
drivers/clocksource/timer-keystone.c:174:10: warning: ignoring return value of 'PTR_ERR', declared with attribute warn_unused_result [-Wunused-result]
   return PTR_ERR(clk);
          ^~~~~~~~~~~~
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/clocksource/timer-keystone.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/drivers/clocksource/timer-keystone.c b/drivers/clocksource/timer-keystone.c
index 7052d85..8932ddc 100644
--- a/drivers/clocksource/timer-keystone.c
+++ b/drivers/clocksource/timer-keystone.c
@@ -147,7 +147,7 @@ static int keystone_set_periodic(struct clock_event_device *evt)
 	return 0;
 }
 
-static void keystone_timer_init(struct device_node *np)
+static int keystone_timer_init(struct device_node *np)
 {
 	struct clock_event_device *event_dev = &timer.event_dev;
 	unsigned long rate;
-- 
1.7.5.4

